main.tf

```diff
@@ -0,0 +1,37 @@
+locals {
+  node_name_suffix = "${var.cluster_id}.${var.base_domain}"
+
+  # Allow sshd access only from loadbalancers
+  node_additional_user_data = {
+    "write_files" = [
+      {
+        path       = "/etc/hosts.allow"
+        "encoding" = "b64"
+        "owner"    = "root:root"
+        "permissions" : "0644"
+
+        "content" = base64encode(format("sshd: %s\n", join(",", module.lb.private_ipv4_addresses)))
+      },
+      {
+        path       = "/etc/hosts.deny"
+        "encoding" = "b64"
+        "owner"    = "root:root"
+        "permissions" : "0644"
+
+        "content" = base64encode("sshd: ALL\n")
+      }
+    ]
+  }
```

Comment:
IMHO, this is not really necessary, as the cluster hosts are not facing the public internet anyway on cloudscale.ch. There's an argument to keep the config to prevent users from hopping between cluster hosts with SSH, but I don't think that's particularly relevant.

Reply:
> Restrict SSH access to cluster VMs to only be possible from LBs.

I did understand it as an acceptance criteria in the ticket. Maybe it is a bit over-engineered and a secret key und private network should be enough.